### PR TITLE
First section appearance performance test

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T179436

Adds a test which, going forward, should raise a flag if our first section appearance performance regresses.

Loads Obama article from test fixture.

Configures a web view controller to fire two events:
- `startTime` when `.atDocumentEnd` fires, which happens when index.html's DOM is inflated, but **before** our sections are loaded
- `firstSectionAppearance` using method outlined [here](https://davidwalsh.name/detect-node-insertion) when first section appears

Measures difference in time between these two events.

Confirmed effectiveness by measuring time it took to load second section after increasing the 'head-start delay' used after first section is loaded. The tests reflected this increase.

### CODE REVIEW

**Comment by montehurd** [comment]
@joewalsh Looks like there's an issue on develop with localized strings. A weird file appears in the Latvian strings folder and compilation hoses...

File: Wikipedia/Code/WebViewController.h [hunk 1]
```diff
@@ -27,6 +28,8 @@ NS_ASSUME_NONNULL_BEGIN
 
 @property (nonatomic) BOOL navBarHidden;
 
+@property (nonatomic, copy, nullable) void (^wkUserContentControllerTestingConfigurationBlock)(WKUserContentController *);
```
Comment (joewalsh) [inline Wikipedia/Code/WebViewController.h]: wrap in `#IF TEST` `#endif`

File: Wikipedia/Code/WebViewController.m [hunk 2]
```diff
@@ -640,6 +640,11 @@ - (WKWebViewConfiguration *)configuration {
                                                    forMainFrameOnly:YES]];
 
     WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
+    
+    if(self.wkUserContentControllerTestingConfigurationBlock) {
```
Comment (joewalsh) [inline Wikipedia/Code/WebViewController.m]: wrap in `#IF TEST` `#endif`

File: Wikipedia/Code/WebViewController.h (hunk 1 above)
Comment (montehurd) [reply Wikipedia/Code/WebViewController.h]: Fixed!

**Comment by montehurd** [comment]
@joewalsh oh weird - the compiler is giving me a warning now in the test file that the method I wrapped in '#if test' doesn't exist...

**Comment by montehurd** [comment]
Re-adding WIP tag 
- tracking down why 'test' isn't set when this test runs
- checking why there seems to be more difference than i'd expect between each of the 10 'measure' runs

**Comment by montehurd** [comment]
Perhaps related to [this](https://stackoverflow.com/a/39435346/135557)?

**Comment by montehurd** [comment]
I found a [way](https://davidwalsh.name/detect-node-insertion) to check for first section appearance without polling! Got it wired up - testing now...

**Comment by montehurd** [comment]
@nambatee I ran into some issues with my test's `measureMetrics` loop's accuracy - finally got them ironed out. I should be able to review/test your (and @mdholloway's) pagelib PRs tomorrow.

**Comment by montehurd** [comment]
@joewalsh @nambatee I think I'm done with tweaks to this :)

**Comment by montehurd** [comment]
Last one :)

**Pull request merged**